### ISSUE TITLE
Scapy sends multiple ARP/ICMP packets when only one is supposed to be sent

### ISSUE THREAD

**Issue by carcigenicate** [secdev/scapy#2831]
#### Brief description

When sending ICMP and ARP packets, multiple are sent when only a single packet is given to `sr`/`sr1`.

#### Environment

-   Scapy version: `2.4.4`
-   Python version: `3.8`
-   Operating System: `Windows 10 64-bit`

I ran this inside of a Pycharm REPL, but the behavior is the same when run as a script on the command line.

#### How to reproduce

```
sr(IP(dst="google.com") / ICMP())
```

Results in:

```
Begin emission:
Finished sending 1 packets.
..*
Received 3 packets, got 1 answers, remaining 0 packets
(<Results: TCP:0 UDP:0 ICMP:1 Other:0>, <Unanswered: TCP:0 UDP:0 ICMP:0 Other:0>)
```

![doubleicmp](https://user-images.githubusercontent.com/18216244/94080865-1f5ba480-fdbc-11ea-9872-2b4ca6b4eb16.jpg)

While this (with a real IP instead of `*`s):

```
sr(ARP(pdst="*.*.*.111"))
```

Results in:

```
Begin emission:
Finished sending 1 packets.
..*
Received 3 packets, got 1 answers, remaining 0 packets
(<Results: TCP:0 UDP:0 ICMP:0 Other:1>, <Unanswered: TCP:0 UDP:0 ICMP:0 Other:0>)
```

![doublearp](https://user-images.githubusercontent.com/18216244/94081965-1a4b2500-fdbd-11ea-9c19-4c036becba5a.jpg)

Then things get odd when using sets of packets. This:

```
ans, _ = sr(IP(dst="google.com", ttl=[1, 2, 3, 4, 5]) / ICMP(), timeout=5)
```

Results in:

```
Begin emission:
Finished sending 5 packets.
.*..*..*..*..*
Received 14 packets, got 5 answers, remaining 0 packets
for _, reply in ans:
    print(reply[IP].src)
 
*.*.*.1
*.*.*.1  # This is the same as the entry above it
*.*.0.1
*.*.0.1  # This is the same as the entry above it
174.0.224.1
```

![multiicmp](https://user-images.githubusercontent.com/18216244/94084246-4b7a2400-fdc2-11ea-8a16-3054e85c2722.png)

Note the duplicate TTLs, and the responses from duplicate addresses. Unless I'm misunderstanding something, these appear to be bugs. 

Sorry for the "redacted" IPs. I don't *think* there's any way those would allow for anything since they're private, but you never know.

#### Expected result

I expect that when I send a single packet, and it indicates in the verbose output that a single packet is sent, that only will be sent. Multiple are being sent though, even when the output indicates that only one was.

**Comment by carcigenicate** [comment]
I figured out that I can fixed the double sending by disabling IP Forwarding in Windows (`Computer\HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Services\Tcpip\Parameters`, `IPEnableRouter` in the Registry).

I noticed that the TTL of the duplicate is lower than the first packet. I'm guessing that two packets are being sent on two interfaces (for some reason), and with `IPEnableRouter` enabled, Windows acts as a router, decrements the TTL, then forwards the packet back onto the same interface as the first packet was sent on? Or something?

**Comment by gpotter2** [comment]
Thanks for the report. This sounds like it's no scapy issue.

My guess is that the name resolution (which is handled by windows through the `socket.gethostbyaddr` call) when using a domain name triggers the ping at some point.

**Comment by carcigenicate** [comment]
@gpotter2 That sounds reasonable in the exact scenario I posted originally. I was able to reproduce it though using:

    sr1(IP(dst="216.58.193.78", ttl=5) / ICMP())

When I tried just now to get evidence though, I can no longer reproduce it! Which is good because that was really messing with my traceroute program. I didn't change anything though besides disabling/re-enabling `IPEnableRouter` and restarting my computer twice.
